lib/docs/filters/reactivex/clean_html.rb

```diff
@@ -0,0 +1,65 @@
+module Docs
+  class Reactivex
+    class CleanHtmlFilter < Filter
+      def call
+        # Can't use options[:container] because then the navigation bar wouldn't be scraped
+        @doc = at_css(root_page? ? '.col-md-8' : '.col-sm-8')
+
+        # Remove breadcrumbs
+        css('.breadcrumb').remove
+
+        # Replace interactive demo's with links to them
+        css('rx-marbles').each do |node|
+          node.name = 'a'
+          node.content = 'Open interactive diagram on rxmarbles.com'
+          node['href'] = "https://rxmarbles.com/##{node['key']}"
+          node.remove_attribute('key')
+        end
+
+        # Syntax-highlighted code blocks
+        css('.code').each do |node|
+          language = node['class'].gsub('code', '').strip
+
+          pre = node.at_css('pre')
+          pre['data-language'] = language
+          pre.content = pre.content.strip
+
+          node.replace(pre)
+        end
+
+        # Assume JavaScript syntax for code blocks not surrounded by a div.code
+        css('pre').each do |node|
+          next if node['data-language']
+
+          node.content = node.content.strip
+          node['data-language'] = 'javascript'
+        end
+
+        # Make language specific implementation titles prettier
+        css('.panel-title').each do |node|
+          # Remove the link, keep the text
+          node.content = node.content
+
+          # Transform it into a header for better styling
+          node.name = 'h3'
+        end
+
+        # Remove language specific implementations that are TBD
+        css('span').each do |node|
+          next unless node.content == 'TBD'
+          node.xpath('./ancestor::div[contains(@class, "panel-default")][1]').remove
+        end
+
+        # Remove the : at the end of "Language-Specific Information:"
+        css('h2').each do |node|
+          node.inner_html = node.inner_html.gsub('Information:', 'Information')
+        end
+
```

Comment:
On http://localhost:9292/reactivex/documentation/operators#categorized, the linked code blocks don’t show up properly. This is because the HTML structure is like `<a><strong><code>...</code></strong></a>`, so the `<code>` element doesn’t know it’s supposed to inherit its color. This could be fixed either by stripping out the extra `<strong>`, moving it outside of the `<a>`, or changing the base DevDocs CSS to match a `<code>` anywhere inside a `<a>`. Here’s the offending rule:

https://github.com/freeCodeCamp/devdocs/blob/6efad323ff55edd81e72a699f55ae5e4d7c80c73/assets/stylesheets/global/_base.scss#L121

Reply:
Looking  at this again, the `<strong>` seems to have some significance in this doc, so going the route of changing the above selector would be better, so as to retain the boldface. However, it seems there’s an additional rule specifying `font-weight: normal` for code spans, so that needs to be either overridden or removed (if it’s not necessary).